### PR TITLE
[GPU] Optimize stable diffusion perf igpu

### PR DESCRIPTION
**Pull request by hyunback**
Improved stable diffusion performance using gemm_tiled_opt by removing gemm transposed input and adding permute.

### Tickets:
 - *104516*

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/ops/matmul.cpp [hunk 1]
```diff
@@ -194,10 +194,13 @@ static void CreateMatMulOp(Program& p, const std::shared_ptr<ngraph::op::v0::Mat
...
-                               [] (const ngraph::Dimension& dim) { return dim.is_static() && dim.get_length() >= 64; }) &&
-                   std::all_of(shapes[1].rbegin(), shapes[1].rbegin() + 2,
-                               [] (const ngraph::Dimension& dim) { return dim.is_static() && dim.get_length() >= 64; });
+            auto is_u8_i8 = (type == ov::element::Type_t::i8 || type == ov::element::Type_t::u8);
+            // Heuristic condition for permute and tiled_opt kernel perform better than ref kernel.
+            return  (std::all_of(shapes[0].rbegin(), shapes[0].rbegin() + 2,
+                                [] (const ngraph::Dimension& dim) { return dim.is_static() && dim.get_length() >= 64; }) &&
+                        std::all_of(shapes[1].rbegin(), shapes[1].rbegin() + 2,
+                                [] (const ngraph::Dimension& dim) { return dim.is_static() && dim.get_length() >= 64; })) ||
+                    ((tensor_from_dims(shapes[0].to_shape()).count() > 100000 || tensor_from_dims(shapes[1].to_shape()).count() > 100000) && !is_u8_i8);
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/ops/matmul.cpp:203]: could you please add test case for this?

Comment (isanghao) [inline src/plugins/intel_gpu/src/plugin/ops/matmul.cpp:203]: What about rewriting the code for readability? for example...
```
bool in0_large = std::all_of(shapes[0].rbegin(), shapes[0].rbegin() + 2,
                                [] (const ngraph::Dimension& dim) { return dim.is_static() && dim.get_length() >= 64; });
bool in1_large = std::all_of(shapes[1].rbegin(), shapes[1].rbegin() + 2,
                                [] (const ngraph::Dimension& dim) { return dim.is_static() && dim.get_length() >= 64; });
bool in0_very_large = tensor_from_dims(shapes[0].to_shape()).count() > 100000;
bool in1_very_large = tensor_from_dims(shapes[0].to_shape()).count() > 100000;
return (in0_large && in1_large) 
             || ((in0_very_large || in1_very_large) && !is_u8_i8);
```

**Review by isanghao — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/src/plugin/ops/matmul.cpp (hunk 1 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/plugin/ops/matmul.cpp]: Added

Comment (hyunback) [reply src/plugins/intel_gpu/src/plugin/ops/matmul.cpp]: Applied

**Review by isanghao — APPROVED** [review]

**Pull request merged**